core/dbt/config/project.py

```diff
@@ -116,6 +116,7 @@ def package_and_project_data_from_root(project_root):
     dependent_projects_dict = {}
     if "packages" in dependencies_yml_dict:
         packages_dict["packages"] = dependencies_yml_dict["packages"]
+        packages_dict["packages_from_dependencies"] = True
```

Comment:
Not related to the change in this PR, as much as the changes previously made in https://github.com/dbt-labs/dbt-core/pull/7857:

This will now start raising an error if any `packages.yml` is an empty file, whether defined in the project root or in an installed package. (For example, the [`dbt-date` package](https://github.com/calogica/dbt-date/blob/main/packages.yml).) That's not exactly supported behavior, but it didn't used to raise an error, and now it does.

```yml
packages:
  - package: calogica/dbt_date
    version: 0.7.2
```
```
$ dbt deps && dbt parse
...
  File "/Users/jerco/dev/product/dbt-core/core/dbt/config/project.py", line 108, in package_and_project_data_from_root
    if "packages" in packages_yml_dict and "packages" in dependencies_yml_dict:
TypeError: argument of type 'NoneType' is not iterable
```

Quick fix on lines 103-106 above:
```python
    if path_exists(package_filepath):
        packages_yml_dict = _load_yaml(package_filepath) or {}
    if path_exists(dependencies_filepath):
        dependencies_yml_dict = _load_yaml(dependencies_filepath) or {}
```

Reply:
Good catch. I've added a test and made this change.